src/Microsoft.AspNet.OData/Batch/ODataBatchContent.cs

```diff
@@ -29,9 +31,29 @@ public partial class ODataBatchContent : HttpContent
         /// <param name="requestContainer">The dependency injection container for the request.</param>
         /// <remarks>This signature uses types that are AspNet-specific.</remarks>
         public ODataBatchContent(IEnumerable<ODataBatchResponseItem> responses, IServiceProvider requestContainer)
+           : this(responses, requestContainer, null /*contentType*/)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataBatchContent"/> class.
+        /// </summary>
+        /// <param name="responses">The batch responses.</param>
+        /// <param name="requestContainer">The dependency injection container for the request.</param>
+        /// <param name="contentType">The response content type.</param>
+        /// <remarks>This signature uses types that are AspNet-specific.</remarks>
+        public ODataBatchContent(IEnumerable<ODataBatchResponseItem> responses, IServiceProvider requestContainer,
+            MediaTypeHeaderValue contentType)
         {
             this.Initialize(responses, requestContainer);
-            Headers.ContentType = MediaTypeHeaderValue.Parse(String.Format(CultureInfo.InvariantCulture, "multipart/mixed;boundary=batchresponse_{0}", Guid.NewGuid()));
+
+            if (contentType == null)
+            {
+                contentType = MediaTypeHeaderValue.Parse(
+                    String.Format(CultureInfo.InvariantCulture, "multipart/mixed;boundary=batchresponse_{0}", Guid.NewGuid()));
```

Comment:
see the same comments in "ODataBatchContent"

Reply:
This is the original file. As mentioned, it is more readable.

---
In reply to: [202868857](https://github.com/OData/WebApi/pull/1536#discussion_r202868857) [](ancestors = 202868857)